### PR TITLE
Insert hidden `#code...code` headings to emulate old readme site

### LINKED ISSUE #6: Support redirecting for URL fragments (page subheadings), especially `code...code` in target fields
From #4:

It looks like the heading fragments differ for some items, e.g. https://www.pantsbuild.org/v2.19/docs/reference-deploy_jar#coderesolvecode links to the resolve field specifically, while the redirect https://docs.pantsbuild.org/2.19.x/reference/targets/deploy_jar#coderesolvecode does not. Can we run client side JS easily (as part of the redirect)? If so, I guess we could replace `codeXYZcode` with `XYZ`, and maybe that handles most cases: the only ones I could find are the field headings for a target.

### PR DESCRIPTION
**Pull request by huonw**
This inserts hidden headings with id `code<alias>code` for each field on a target, to bridge the difference between the fragment anchors used on the old side and the ones on the new site. This means an old link to a specific field will be redirected to the new page and then land on the appropriate field.

This has to do a bit of hacks to have this work "perfectly", and in particular have the field heading shown and not hidden by the sticky nav bar. In particular, get this to work, we have to use real markdown headings (`#######`), but we want them hidden both in the table of contents and on the page itself.

Comparison of the two styles of headings:
- old: https://www.pantsbuild.org/docs/reference-python_requirements#codemodule_mappingcode
- new: https://docs.pantsbuild.org/2.18/reference/targets/python_requirements#module_mapping

Fixes #6, using the idea https://github.com/pantsbuild/pantsbuild.org/issues/6#issuecomment-1872647373.

Commits are broken up sensibly.

### CODE REVIEW

**Comment by huonw** [comment]
WIP because it'll be nicest to merge this once we've done the major regeneration (#50) and thus the regeneration here is focused.

**Comment by thejcannon** [comment]
Oddly, when I tried this, it made it scroll past the header a bit:

http://localhost:3000/2.20/reference/targets/deploy_jar#tags
Screenshot from 2024-01-03 21-03-59

http://localhost:3000/2.20/reference/targets/deploy_jar#codetagscode
Screenshot from 2024-01-03 21-05-01

Funny enough if I move the `<a>` tag above the h2, it looks also scrolled below (albeit a few pixels higher)

---

I suspect we might have to make it so the reference docs for older versions use [explicit heading IDs](https://docusaurus.io/docs/markdown-features/toc#heading-ids)

**Comment by huonw** [comment]
Yeah, that's:

> it's not perfect, as the heading seems to end up off the screen, but it's good enough

My take was it's good enough... although an explicit heading ID is an interesting idea that I didn't know about.

It looks like that would have the downside of causing the new site to have inconsistent fragments between versions, so switching between versions won't necessarily land on the right spot. I think this is worse (on balance) than old links being slightly wrong, as we'll "always" be stuck with that problem, rather that it being a "temporary" headache.

I'll do some experiments.

**Comment by huonw** [comment]
Okay, I got to a middle ground that seems to work: headings with explicit heading IDs, that are hidden via CSS/`aria-hidden`.

File: docs/reference/targets/adhoc_tool.mdx [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ---
 title: adhoc_tool
+# The old-style `code...code` anchors are heading level 6, and shouldn't be visible in ToC
```
Comment (thejcannon) [inline docs/reference/targets/adhoc_tool.mdx:3]: Did you see/ did you try just using h tags so they don't show up in ToC?

https://docusaurus.io/docs/markdown-features/toc#customizing-table-of-contents-generation

Comment (huonw) [reply docs/reference/targets/adhoc_tool.mdx:3]: Yep, it has the same positioning problem as the original version. You can see the minor variations of this current state described in the big comment in the field template: https://github.com/pantsbuild/pantsbuild.org/pull/55/commits/6bb8ab687de358fb8c7332d858e00b18c8a4c03c

(I note the comment refers to `<A />` but the same applies to any HTML-tag-based method.)

**Comment by thejcannon** [comment]
I opened https://github.com/pantsbuild/pantsbuild.org/pull/72 to see if maybe we could do 4 from https://github.com/pantsbuild/pantsbuild.org/issues/6#issuecomment-1872622556 now knowing that the "fake IDs" suggestion isn't as seamless as we'd hoped.

**Comment by huonw** [comment]
Closing in favour of #72.

**Pull request closed without merging**